flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java

```diff
@@ -176,6 +177,10 @@ public FlinkKinesisConsumer(List<String> streams, KinesisDeserializationSchema<T
 		// check the configuration properties for any conflicting settings
 		KinesisConfigUtil.validateConsumerConfiguration(this.configProps);
 
+		checkArgument(
```

Comment:
How about first checking for null, and then for serializability? (Just seem more intuitive to me)